File: ga/4omini_result_2024_7/result/1968B_mini/37_1968B.c
use std::io;

fn main() {
    let stdin = io::stdin();
    let mut input = String::new();

    // Read the number of test cases
    stdin.read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();
    
    for _ in 0..t {
        input.clear();
        
        // Read n, m, sn, and sm
        stdin.read_line(&mut input).unwrap();
        let parts: Vec<&str> = input.split_whitespace().collect();
        let n: usize = parts[0].parse().unwrap();
        let m: usize = parts[1].parse().unwrap();
        
        let sn = parts[2];
        let sm = parts[3];

        let mut i = 0;
        let mut j = 0;

        // Count how many characters from sn can be matched in sm
        while i < n && j < m {
            if sn.as_bytes()[i] == sm.as_bytes()[j] {
                i += 1;
            }
            j += 1;
        }

        // Output the count of matched characters
        println!("{}", i);
    }
}

File: ga/gpt3_5_result_2024_6/result/1968B_gpt/37_1968B.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: i32 = input.trim().parse().unwrap();
    
    for _ in 0..t {
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let values: Vec<i32> = input
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();
        
        let n = values[0];
        let m = values[1];
        
        let mut sn = String::new();
        io::stdin().read_line(&mut sn).unwrap();
        sn = sn.trim().to_string();
        
        let mut sm = String::new();
        io::stdin().read_line(&mut sm).unwrap();
        sm = sm.trim().to_string();
        
        let mut i = 0;
        let mut j = 0;
        
        while i < n && j < m {
            if sn.chars().nth(i) == sm.chars().nth(j) {
                i += 1;
            }
            j += 1;
        }
        
        println!("{}", i);
    }
}